src/parser/basic.rs

```diff
@@ -0,0 +1,45 @@
+//! Default [`Parser`] implementation.
+
+use std::{path::Path, vec};
+
+use futures::stream;
+
+use crate::Parser;
+
+/// Default [`Parser`].
+///
+/// As there is no async runtime-agnostic way to interact with io, this
+/// [`Parser`] is blocking.
+#[derive(Clone, Copy, Debug)]
+pub struct Basic;
+
+impl<I: AsRef<Path>> Parser<I> for Basic {
+    type Output = stream::Iter<vec::IntoIter<gherkin::Feature>>;
+
+    fn parse(self, path: I) -> Self::Output {
+        let path = path
+            .as_ref()
+            .canonicalize()
+            .expect("failed to canonicalize path");
```

Comment:
Don't like that. It's better to make the trait fallible and async (for future proofing, while the current implementatio still can be blocking). And it seems that the `Stream::Item` should be fallible. I think using `io::Result` will quite enough for that.

Once trait is fallible, we may reason about errors on the level above, which is much better.

Reply:
As we return [`Stream`](https://docs.rs/futures/0.3.16/futures/stream/trait.Stream.html) current trait is `async`-ready. I think it's better to hide futures with `async { ... }`[`.into_stream()`](https://docs.rs/futures/0.3.16/futures/future/trait.FutureExt.html#method.into_stream)[`.skip(1)`](https://docs.rs/futures/0.3.16/futures/stream/trait.StreamExt.html#method.skip)[`.chain(features)`](https://docs.rs/futures/0.3.16/futures/stream/trait.StreamExt.html#method.chain) inside implementation, than calling `.await` twice with something like `parser.parse().await.collect().await`

Regardless making `Stream::Item` being fallible I agree, that's a great idea.